### ISSUE TITLE
Lucifer cannon blast doesn't depend on charge level

### ISSUE THREAD

**Issue by lamefun** [Unvanquished/Unvanquished#589]
Even if you only charged 20% blast, it still looks like a 100% blast.

**Comment by Viech** [comment]
True. The particle system doesn't allow scaling based on a gamelogic value. This is a desirable feature though.

**Comment by bmorel** [comment]
Maybe a workaround for that would be to modify colour of particles?

**Comment by Viech** [comment]
That is also part of either the asset or the shader, so the same limitations as for particle scale should apply. A good enough workaround would be to have few intermediate particle systems and choose the appropriate one client side. The missile is spawned server side, so it should be possible to tell the client how much damage it does.

**Comment by necessarily-equal** [comment]
The very short (right click) blast is displayed differently. I'm not sure if it's a different kind of missile or not, but I would definitely take inspiration here.

Having a look, it seems that `src/cgame/cg_weapon.cpp`'s `FireLcannonHelper` does save the charge in `(gentity_t* missile)->s.torsoAnim` but that the `MIS_LCANNON`/`MIS_LCANNON2` difference does the shader magic. (`bg_missilesData` assigns them to the `lcannon` and `lcannon2` names)

**Comment by Viech** [comment]
Yes, this is what I expected. One would implement it as if the luci had three or more distinguished fire modes.
